Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -354,6 +354,10 @@ synchronized private static Branch initBranchSDK(@NonNull Context context, Strin
             branchReferral_.prefHelper_.setBranchKey(PrefHelper.NO_STRING_VALUE);
         } else {
             branchReferral_.prefHelper_.setBranchKey(branchKey);
+            // Set the source to "public_function" since this method is called via getAutoInstance with explicit key
+            if (!branchKey.equals(BranchUtil.readBranchKey(context))) {
+                branchReferral_.prefHelper_.setBranchKeySource("public_function");
```

Comment:
I would change to "init_function", both this and `setBranchKey` are public functions